document_page_approval_levels/security/res_groups.xml

```diff
@@ -14,5 +14,16 @@
             <field name="implied_ids" eval="[(4, ref('group_document_approver'))]"/>
         </record>
 
+       <record id="group_document_user_rule" model="ir.rule">
+            <field name="name">Can only delete your document</field>
+            <field name="model_id" ref="document_page.model_document_page"/>
+            <field name="domain_force">['|', ('create_uid', '=', user.id), ('create_uid', '=', False)]</field>
+            <field name="groups" eval="[(4, ref('base.group_document_user'))]" />
+            <field name="perm_read" eval="0"/>
```

Comment:
you need make first the read in document before make unlink

what this rule is working?